lib/views/pages/create_account_page.dart

```diff
@@ -17,20 +17,20 @@ class CreateAccountPage extends HookConsumerWidget {
   Widget build(BuildContext context, WidgetRef ref) {
     navigateToLoginPageOnLogout(context, ref);
 
-    return Scaffold(
-      appBar: AppBar(
-        leading: IconButton(
-          key: logoutButtonKey,
-          onPressed: () {
-            ref.read(loginServiceProvider).signOut();
-          },
-          icon: const Icon(Icons.arrow_back),
+    return PopScope(
+      canPop: false,
+      onPopInvoked: (bool didPop) {
+        ref.read(loginServiceProvider).signOut();
+      },
+      child: Scaffold(
+        appBar: AppBar(
+          leading: const LeadingBackButton(),
```

Comment:
Why was it chosen to use a PopScope, instead of the callback from LeadingBackButton?

Reply:
That way it also handle and cancel the back arrow on android phone (to make sure that we go back to the login back only if we are log out) and keep the `LeadingBackButton` be a more general widget that is just there to pop the current page.

Reply:
Oh, that's interesting! Good idea and good job :)